os/arch/arm/src/imxrt/imxrt_qtmr.c

```diff
@@ -1,563 +1,1019 @@
- * param mask     The DMA to enable. This is a logical OR of members of the
- *                  enumeration ::qtmr_dma_enable_t
- */
-void imxrt_qtmr_enabledma(TMR_Type *base, qtmr_channel_selection_t channel, uint32_t mask)
-{
-	uint16_t reg;
-
-	reg = base->CHANNEL[channel].DMA;
-	/* Input Edge Flag DMA Enable */
-	if (mask & kQTMR_InputEdgeFlagDmaEnable) {
-		/* Restriction: Do not set both DMA[IEFDE] and SCTRL[IEFIE] */
-		base->CHANNEL[channel].SCTRL &= ~TMR_SCTRL_IEFIE_MASK;
-		reg |= TMR_DMA_IEFDE_MASK;
-	}
-	/* Comparator Preload Register 1 DMA Enable */
-	if (mask & kQTMR_ComparatorPreload1DmaEnable) {
-		reg |= TMR_DMA_CMPLD1DE_MASK;
-	}
-	/* Comparator Preload Register 2 DMA Enable */
-	if (mask & kQTMR_ComparatorPreload2DmaEnable) {
-		reg |= TMR_DMA_CMPLD2DE_MASK;
-	}
-	base->CHANNEL[channel].DMA = reg;
-}
-
-/*!
- * brief Disable the Quad Timer DMA.
- *
- * param base     Quad Timer peripheral base address
- * param channel  Quad Timer channel number
- * param mask     The DMA to enable. This is a logical OR of members of the
- *                  enumeration ::qtmr_dma_enable_t
- */
-void imxrt_qtmr_disabledma(TMR_Type *base, qtmr_channel_selection_t channel, uint32_t mask)
-{
-	uint16_t reg;
-
-	reg = base->CHANNEL[channel].DMA;
-	/* Input Edge Flag DMA Enable */
-	if (mask & kQTMR_InputEdgeFlagDmaEnable) {
-		reg &= ~TMR_DMA_IEFDE_MASK;
-	}
-	/* Comparator Preload Register 1 DMA Enable */
-	if (mask & kQTMR_ComparatorPreload1DmaEnable) {
-		reg &= ~TMR_DMA_CMPLD1DE_MASK;
-	}
-	/* Comparator Preload Register 2 DMA Enable */
-	if (mask & kQTMR_ComparatorPreload2DmaEnable) {
-		reg &= ~TMR_DMA_CMPLD2DE_MASK;
-	}
-	base->CHANNEL[channel].DMA = reg;
-}
+/****************************************************************************
+ *
+ * Copyright 2019 NXP Semiconductors All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/*
+ * Copyright 2017 NXP
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include <debug.h>
+#include <errno.h>
+
+#include <tinyara/clock.h>
+#include <tinyara/timer.h>
+
+#include "imxrt_clock.h"
+#include "imxrt_qtmr.h"
+#if defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT102x)
+#include "imxrt102x_irq.h"
+#elif defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT105x)
+#include "../../include/imxrt/imxrt105x_irq.h"
```

Comment:
I think you can use below instead of above.
#include <tinyara/irq.h>

Reply:
Ok. Done.